src/V3Undriven.cpp

```diff
@@ -151,7 +151,18 @@ class UndrivenVarEntry final {
     void reportViolations() {
         // Combine bits into overall state
         AstVar* const nodep = m_varp;
-        {
+
+        if (nodep->isGenVar()) { // Genvar
+            if (!nodep->isIfaceRef() && !nodep->isUsedParam() && !unusedMatch(nodep)) {
+                nodep->v3warn(UNUSEDGENVAR, "Genvar is not used: " << nodep->prettyNameQ());
```

Comment:
The alternative way which is perhaps cleaner here and elsewhere is to just have UNUSED be a normal warning then instead where we warn test both are turned on:

```suggestion
                if (!nodep->fileline()->warnIsOff(V3ErrorCode::UNUSED) {
                   nodep->v3warn(UNUSEDGENVAR, "Genvar is not used: " << nodep->prettyNameQ());
                }
```

Reply:
So, if I am understanding this correctly, you want to have both, but `UNUSED` has higher priority over the individual unused ones (`UNUSEDGENVAR`, `UNUSEDPARAM`, `UNUSEDSIGNAL`)? How would I handle this command line:

`-Wall -Wno-UNUSEDGENVAR`

1. `-Wall` enables both `UNUSED`  and `UNUSEDGENVAR` warnings
2. `-Wno-UNUSEDGENVAR` disables `UNUSEDGENVAR` warning, but `UNUSED` warning is still active which means `UNUSEDGENVAR` warning will still be shown.

Reply:
This will not work properly when enabling one of the two `-Wwarn-UNUSEDGENVAR` or `-Wwarn-UNUSED` .

This will only work when having everything enabled using for example `-Wall` and then disabling individual ones with `-Wno-`. Also, I believe `-Werror-UNUSED` will not work either.

The way I did should handle every case.

Reply:
Reasonable point, ok keep as is.  Note tests still need to be passing.